### PR TITLE
Expand documentation and TODOs in a few packages

### PR DESCRIPTION
**Pull request by jbeda**
I was reading through unfamiliar code and mostly added TODOs and expanded and clarified documentations.

There are a couple of things that are real code changes:
- Removed some unused constants
- Changed `workqueue.Parallize` to clamp the number of worker goroutines to the number of items to be processed.
- Added another unit test to `workqueue.queue`.  I thought I found a bug (I was wrong) and wrote a unit test to isolate.  I figure the extra test is worth keeping.

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/35382)

### CODE REVIEW

File: pkg/util/wait/wait.go
Comment (brendandburns) [inline pkg/util/wait/wait.go]: space between sliding and is

Comment (brendandburns) [inline pkg/util/wait/wait.go]: And everyone always yells at me about double-space after period, even though it feels natural to me too...

File: pkg/util/workqueue/rate_limitting_queue.go
Comment (brendandburns) [inline pkg/util/workqueue/rate_limitting_queue.go]: This sentence doesn't parse for me.

**Review by brendandburns — DISMISSED** [review]

File: pkg/util/wait/wait.go
Comment (jbeda) [inline pkg/util/wait/wait.go]: *sigh* fixing.  I should add something to https://github.com/alecthomas/gometalinter.

Comment (jbeda) [inline pkg/util/wait/wait.go]: But this is really common in our codebase...

```
$ find . \( -path './vendor' -o -path './third_party' -o -path './staging' \) -prune -o -name '*.go' -print | xargs grep '\.  ' | wc -l
    2094
```

**Comment by jbeda** [comment]
@brendandburns PTAL

File: pkg/util/wait/wait.go
Comment (jbeda) [inline pkg/util/wait/wait.go]: `golint` didn't flag this.  In fact, this format is supported by `gofmt` and is used as an example in [Effective Go](https://golang.org/doc/effective_go.html#formatting).

I'm happy to change, but this seems a bit pedantic.

**Comment by kargakis** [comment]
If golint didn't complain then that's fine

**Comment by brendandburns** [comment]
LGTM.

**Comment by brendandburns** [comment]
@jbeda looks like a flake?  Does this pass unit/integration on your machine?

**Comment by jbeda** [comment]
@brendandburns I think it was a flake.  Rebased and repushed.  Also, you need to approve the github review, I think.

**Comment by jbeda** [comment]
@brendandburns This looks green now  Just waiting on LGTM and approve from you.

**Comment by brendandburns** [comment]
LGTM again.

**Comment by jbeda** [comment]
Rebased -- please re-LGTM.

**Comment by saad-ali** [comment]
Release-czar approved post-code freeze merge--This was LGTMed and in the merge-queue at code freeze time for 1.5. Adding 1.5 milestone to let it gets merged after code freeze.

**Comment by dims** [comment]
@k8s-bot unit test this

**Pull request merged**
